examples/pxScene2d/src/rcvrcore/optimus.js

```diff
@@ -482,6 +482,9 @@ function Optimus() {
    * @returns {Application}
    */
   this.createApplication = function(props){
+    if (this.getApplicationById(props.id) !== null) {
+      throw new Error("An application with ID '" + props.id + "' already exists");
+    }
```

Comment:
Let's not throw an exception here.  Instead, we should still create the app but reject its ready promise.  Also, we should still return the app but not add the app to the applicationsArray.